CIME/baselines/performance.py

```diff
@@ -80,16 +80,17 @@ def perf_compare_memory_baseline(case, baseline_dir=None):
         baseline_dir = case.get_baseline_dir()
 
     config = load_coupler_customization(case)
-
-    baseline_file = os.path.join(baseline_dir, "cpl-mem.log")
-
-    baseline = read_baseline_file(baseline_file)
-
     tolerance = case.get_value("TEST_MEMLEAK_TOLERANCE")
 
     if tolerance is None:
         tolerance = 0.1
 
+    baseline_file = os.path.join(baseline_dir, "cpl-mem.log")
+    if not os.path.exists(baseline_file):
+        return tolerance, "MEMCOMP: No baseline memory file present"
```

Comment:
@jgfouca I would like this string to appear in the TestStatus file

Reply:
@jedwards4b , I think you need to return a boolean here, right?
`return True, "MEMCOMP: No baseline memory file present"`

I'm a little worried that this will hide errors. I would not have noticed that our perf test was not running long enough to create a cpl-mem file if it was passing. @jasonb5 , thoughts?

Reply:
Yes it should be boolean.  The options for TEST_*_STATUS are PASS, FAIL and PEND - should I add a TEST_SKIP_STATUS?

Reply:
Just regular tests and I am only dealing with this in system_tests_common.py so I don't think there will be a problem for your performance tests.  Now have:
`PASS SMS_Ln9_Vnuopc.ne30_ne30_mg17.FW2000climo.derecho_intel.cam-outfrq9s MEMCOMP: No baseline memory file present`
in TestStatus file.

Reply:
I agree hiding this could mask true issues.
I wonder if a better way would be adding a new flag when creating the baseline that would populate something to inform future tests that memory checking can be skipped. This way when the user creates the baseline they understand that memory checking will be skipped rather than it be implicit and it documents this in the baseline for future users that may not have created the baseline.

Reply:
Honestly I just need to bypass this test so that our test suite works for now, can we open a new issue to implement your suggested change?  I just don't have the bandwidth to do that right now.

Reply:
Maybe just skip the performance stuff if SAVE_TIMING is off?

Reply:
I'm also wondering why we didn't see this in any of our tests? A lot of our tests are short running.

Reply:
This test runs 9 timesteps - do you have tests that short?  If so please look into this.   Thanks,

Reply:
I see, our script doesn't have that capability and we are capturing a bunch of test failures that we should not be capturing.

Reply:
Does that script live in CIME?

Reply:
no I believe that it's outside of cime @fischer-ncar?

Reply:
It's not a script.  It's users looking at TestStatus and seeing that cpl-mem.log is missing.  Then they start contacting me asking what went wrong.

Reply:
I just remembered, a lot of people use cs.status.

Reply:
what dies the output of cs.status look like when cpl-mem.log is missing?

Reply:
ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s (Overall: PASS) details:
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s CREATE_NEWCASE
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s XML
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s SETUP
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s SHAREDLIB_BUILD time=143
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s NLCOMP
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s MODEL_BUILD time=238
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s SUBMIT
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s RUN time=159
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s COMPARE_base_rest
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s GENERATE cesm2_3_alpha17c
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s BASELINE cesm2_3_alpha17b:
    FAIL ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s MEMCOMP [Errno 2] No such file or directory: '/glade/campaign/cesm/cesmdata/cesm_baselines/cesm2_3_alpha17b/ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s/cpl-mem.log'
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s TPUTCOMP
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s MEMLEAK insufficient data for memleak test
    PASS ERP_D_Ln9_Vnuopc.f19_f19_mg17.QPC6.derecho_intel.cam-outfrq9s SHORT_TERM_ARCHIVER

Reply:
This is similar to what happens when the baseline comparison fails, because of missing baselines. So I'm OK with having it marked as FAIL. So far I've been ignoring the MEM and TPUTCOMP tests because we found we couldn't count on them. So I'm glad they are being worked on. It would be nice to be able to use them. So thanks for working on this.